boilerplate/app/components/Button.tsx

```diff
@@ -181,53 +184,54 @@ const $baseViewStyle: ViewStyle = {
   paddingVertical: spacing.sm,
   paddingHorizontal: spacing.sm,
   overflow: "hidden",
-}
+})
 
-const $baseTextStyle: TextStyle = {
+const $baseTextStyle: ThemedStyle<TextStyle> = ({ typography }) => ({
   fontSize: 16,
   lineHeight: 20,
   fontFamily: typography.primary.medium,
   textAlign: "center",
   flexShrink: 1,
   flexGrow: 0,
   zIndex: 2,
-}
-
-const $rightAccessoryStyle: ViewStyle = { marginStart: spacing.xs, zIndex: 1 }
-const $leftAccessoryStyle: ViewStyle = { marginEnd: spacing.xs, zIndex: 1 }
-
-const $viewPresets = {
+})
+
+const $rightAccessoryStyle: ThemedStyle<ViewStyle> = ({ spacing }) => ({
+  marginStart: spacing.xs,
+  zIndex: 1,
+})
+const $leftAccessoryStyle: ThemedStyle<ViewStyle> = ({ spacing }) => ({
+  marginEnd: spacing.xs,
+  zIndex: 1,
+})
+
+const $viewPresets: Record<Presets, ThemedStyleArray<ViewStyle>> = {
   default: [
     $baseViewStyle,
-    {
+    ({ colors }) => ({
       borderWidth: 1,
       borderColor: colors.palette.neutral400,
       backgroundColor: colors.palette.neutral100,
-    },
-  ] as StyleProp<ViewStyle>,
-
-  filled: [$baseViewStyle, { backgroundColor: colors.palette.neutral300 }] as StyleProp<ViewStyle>,
-
-  reversed: [
-    $baseViewStyle,
-    { backgroundColor: colors.palette.neutral800 },
-  ] as StyleProp<ViewStyle>,
+    }),
+  ],
+  filled: [$baseViewStyle, ({ colors }) => ({ backgroundColor: colors.palette.neutral300 })],
+  reversed: [$baseViewStyle, ({ colors }) => ({ backgroundColor: colors.palette.neutral800 })],
 }
 
-const $textPresets: Record<Presets, StyleProp<TextStyle>> = {
-  default: $baseTextStyle,
-  filled: $baseTextStyle,
-  reversed: [$baseTextStyle, { color: colors.palette.neutral100 }],
+const $textPresets: Record<Presets, ThemedStyleArray<TextStyle>> = {
+  default: [$baseTextStyle],
+  filled: [$baseTextStyle],
+  reversed: [$baseViewStyle, ({ colors }) => ({ color: colors.palette.neutral100 })],
```

Comment:
found a bug in our sample app - this should be changed: 
```suggestion
  reversed: [$baseTextStyle, ({ colors }) => ({ color: colors.palette.neutral100 })],
```

Reply:
Fixed in 9fe7419